web/src/components/Header.js

```diff
@@ -3,62 +3,134 @@ import Button from '@cmsgov/design-system-core/dist/components/Button/Button';
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
 import { Link } from 'react-router-dom';
 import { push } from 'connected-react-router';
+import OutsideClickHandler from 'react-outside-click-handler';
 
-
+import { getIsAdmin } from '../reducers/user';
 import { t } from '../i18n';
 
+import Icon, { faChevronDown, faChevronLeft, faChevronUp, faEdit, faSignOutAlt } from './Icons';
+
 class Header extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ariaExpanded: props.ariaExpanded };
+  }
+
   handleLogout = e => {
     e.preventDefault();
     const { pushRoute } = this.props;
     pushRoute('/logout');
   };
 
+  toggleDropdown = () => {
+    this.setState(prev => ({ ariaExpanded: !prev.ariaExpanded }));
+  };
+
   render() {
-    const { authenticated } = this.props;
+    const { authenticated, currentUser, isAdmin, location } = this.props;
+    const { ariaExpanded } = this.state;
+    const username = currentUser ? currentUser.username : 'Your account';
+    const isTopLevel = location.pathname == '/';
     return (
-      <header className="header__slim">
+      <OutsideClickHandler
```

Comment:
This is the Airbnb handler for this. Looking at the source, it seemed like it did exactly what I was going to write from scratch anyway. https://github.com/airbnb/react-outside-click-handler

Reply:
This is my only concern with the Airbnb library:  https://github.com/airbnb/react-outside-click-handler/issues/24

Reply:
Hrm. Good catch. I suppose I could just write my own!

Reply:
Did you know https://next.material-ui.com/components/click-away-listener/?

Reply:
Hi @oliviertassinari! We're actually in the process of removing Material-UI from the project. Thanks for the info, though!